Project-1/README.md

```diff
@@ -67,9 +183,99 @@ Note: The standard interpretation of the logical symbols - "∨", "∧", "→",
   (b) ∀x∀y(¬(Px ∨ Qx) → (¬Px ∧ ¬Qx))
   (c) ∀x∀y((¬Px ∨ ¬Qx) → ¬(Px ∧ Qx))
   (d) ∀x∀y((¬Px ∧ ¬Qx) → ¬(Px ∨ Qx))
+  
+   (a) ∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
+  
+ (1) ¬∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
+ (2) ¬∀y(¬(Pa ∧ Qa) → (¬Pa ∨ ¬Qa))
+ (3) ¬(¬(Pa ∧ Qa) → (¬Pa ∨ ¬Qa))
+ (4)¬(Pa ∧ Qa)
+ (5)¬(¬Pa ∨ ¬Qa)
+ (6).¬¬Pa
+ (7)¬¬Qa
+ (8)Qa
+ (9) Pa
+ (10) ¬Pa	(11) ¬Qa
+        x	       x
+ 
+  (b) ∀x∀y(¬(Px ∨ Qx) → (¬Px ∧ ¬Qx))
+  
+  (1) ¬∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
+  (2) ¬∀y(¬(Pa ∧ Qa) → (¬Pa ∨ ¬Qa))
+  (3) ¬(¬(Pa ∧ Qa) → (¬Pa ∨ ¬Qa))
+  (4) ¬(Pa ∧ Qa)
+  (5) ¬(¬Pa ∨ ¬Qa)
+  (6) ¬¬Pa
+  (7) ¬¬Qa
+  (8) Qa
+  (9) Pa
+  (10) ¬Pa	(11) ¬Qa
+        x	      x
+  
+  (c) ∀x∀y((¬Px ∨ ¬Qx) → ¬(Px ∧ Qx))
+  
+  (1) ¬∀x∀y((¬Px ∨ ¬Qx) → ¬(Px ∧ Qx))
+  (2) ¬∀y((¬Pa ∨ ¬Qa) → ¬(Pa ∧ Qa))
+  (3) ¬((¬Pa ∨ ¬Qa) → ¬(Pa ∧ Qa))
+  (4) ¬Pa ∨ ¬Qa
+  (5) ¬¬(Pa ∧ Qa)
+  (6) Pa ∧ Qa
+  (7) Pa
+  (8) Qa
+  (9) ¬Pa	(10) ¬Qa
+     x		     x
+  
+  (d) ∀x∀y((¬Px ∧ ¬Qx) → ¬(Px ∨ Qx))
+  
+ (1) ¬∀x∀y((¬Px ∧ ¬Qx) → ¬(Px ∨ Qx))
+ (2) ¬∀y((¬Pa ∧ ¬Qa) → ¬(Pa ∨ Qa))
+ (3) ¬((¬Pa ∧ ¬Qa) → ¬(Pa ∨ Qa))
+ (4) ¬Pa ∧ ¬Qa
+ (5) ¬¬(Pa ∨ Qa)
+ (6) Pa ∨ Qa
+ (7) ¬Pa
+ (8) ¬Qa
+ (9) Pa		(10) Qa
+     x		     x
+     
 ```
 	
 9. Using a natural deduction proof generator - such as the one found here `https://proofs.openlogicproject.org/` - provide natural deduction proofs for each of De Morgan's laws. 
 
+¬∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
+(1) ¬(Px ∧ Qx)		assumption
+(2) ¬(¬Px ∨ ¬Qx)	assumption
+(3) (Px ∧ Qx)		2, negated disjunction
+(4) ¬(Px ∧ Qx)		1,3 contradiction
+
+∀x∀y(¬(Px ∨ Qx) → (¬Px ∧ ¬Qx))
+(1) ¬(Px ∨ Qx)		assumption
+(2) (¬Px ∧ ¬Qx)		assumption
+(3) (Px ∨ Qx)		2, negated conjunction
+(4) ¬(Px ∨ Qx)		1,3 contradiction
+
+∀x∀y((¬Px ∨ ¬Qx) → ¬(Px ∧ Qx))
+(1) (¬Px ∨ ¬Qx)		assumption
+(2) ¬¬(Px ∧ Qx)		assumption
+(3) (Px ∧ Qx)		2, double negation
+(4) Px			3, conjunction
+(5) Qx			3, conjunction
+(6) (¬Px ∨ ¬Qx)		4,5,6 contradiction
+
+∀x∀y((¬Px ∧ ¬Qx) → ¬(Px ∨ Qx))
+(1) (¬Px ∧ ¬Qx)		assumption
+(2) ¬¬(Px ∨ Qx)		assumption
+(3) (Px ∨ Qx)		2, double negation
+(4) ¬Px			1, conjunction
+(5) ¬Qx			1, conjunction
+(6) Px			3, disjunction
+(7) ¬Qx			3, disjunction
+(8) ¬Px			3, disjunction
+(9) Qx			3, disjunction
+(10) (¬Px ∧ ¬Qx)	6,7,8,9 contradiction
```

Comment:
Contradictions occur between two lines, i.e. what is this contradicting?

Reply:
∀x∀y((¬Px ∧ ¬Qx) → ¬(Px ∨ Qx))
(1) (¬Px ∧ ¬Qx)		assumption
(2) ¬¬(Px ∨ Qx)		assumption
(3) (Px ∨ Qx)		2, double negation
(4) ¬Px			1, conjunction
(5) ¬Qx			1, conjunction
(6) Px			3, disjunction
(7) ¬Qx			3, disjunction
(8) ¬Px			3, disjunction
(9) Qx			3, disjunction
(10) (¬Px ∧ ¬Qx)	6,7,8,9 contradiction

This is messy but I think it's the same problem from line 6 of the proof on line 263 in Github.  I'm not sure what I'm doing in this proof but I seem to be deconstructing parts of the sentence to piece back together a disjunction or conjunction that creates a contradiction of one of my first assumptions.  Is there some basic logic rule that I don't know about that I am following here?
I'm not even sure what contradiction I'm pointing to except that in this proof (and in the previous proof that includes like 263) I have also "shown" (maybe) that Px and Qx are true.  Therefore, it is impossible for (¬Px ∧ ¬Qx) to be true - both conjuncts must be true.  I think I've tried to get at this in the proof?

Reply:
Another thought: Maybe I skipped a step and should have included (Px ∧ Qx) as a conjunction in a separate line, that way I can then write (¬Px ∧ ¬Qx) 10, 11 as a contradiction?  I always feel like I'm making up my own rules.

Reply:
I guess I could combine (Px ∧ Qx) once Px and Qx have been proven true.  That's the nature of a conjunction.